spring/spring-data/src/main/java/io/quarkus/ts/spring/data/rest/BeanResource.java

```diff
@@ -0,0 +1,142 @@
+package io.quarkus.ts.spring.data.rest;
+
+import java.util.List;
+
+import jakarta.enterprise.inject.spi.CDI;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InjectableBean;
+import io.quarkus.ts.spring.data.di.Account;
+import io.quarkus.ts.spring.data.di.AccountService;
+import io.quarkus.ts.spring.data.di.AudioBookService;
+import io.quarkus.ts.spring.data.di.BookService;
+import io.quarkus.ts.spring.data.di.PersonDao;
+import io.quarkus.ts.spring.data.di.SpringBeansConfig;
+import io.quarkus.ts.spring.data.di.SpringMainConfig;
+import io.quarkus.ts.spring.data.di.SpringPersonService;
+import io.quarkus.ts.spring.data.di.UserService;
+
+@Path("/bean")
+public class BeanResource {
+
+    private static List<String> beanNameProvider() {
+        return List.of(
+                "account",
+                "accountServiceImpl",
+                "audioBookServiceGenerator",
+                "bookServiceGenerator",
+                "personDaoImpl",
+                "springPersonService",
+                "userService");
+    }
+
+    private static List<Class<?>> beanClassProvider() {
+        return List.of(
+                Account.class,
+                AccountService.class,
+                AudioBookService.class,
+                BookService.class,
+                PersonDao.class,
+                SpringPersonService.class,
+                UserService.class);
+    }
+
+    @GET
+    @Path("/beanExists")
+    @Produces("text/plain")
+    public boolean beanExists() {
+        for (Class<?> beanClass : beanClassProvider()) {
+            final Object bean = CDI.current().select(beanClass).get();
+            if (bean == null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @GET
+    @Path("/givenAccountServiceAutowiredToUserService")
+    @Produces("text/plain")
+    public boolean givenAccountServiceAutowiredToUserService_WhenGetAccountServiceInvoked_ThenReturnValueIsNotNull() {
+        UserService userService = CDI.current().select(UserService.class).get();
+        return userService.getAccountService() != null;
+    }
+
+    @GET
+    @Path("/givenPersonDaoAutowiredToSpringPersonServiceBySetterInjection")
```

Comment:
Few general notes to path:

- Strictly speaking camel case is sometimes used for paths, but IMHO it is extraordinary. For example Google recommends hyphnes https://developers.google.com/search/docs/crawling-indexing/url-structure please prefer `green-apple` to `greenApple`
- your method name is spot on and I'm quite sure that some reviewers of our team would prefer it much more than to what I'm going to suggest, but it is very long and it took me while to figure what it means. Can we simplify it so that when I browse through this resource, it is clear that one method tests SpringPersonService, second test UserService
- I'd like to follow conventions regarding REST naming, this is random page I found https://restfulapi.net/resource-naming/ please don't follow it word by word, but found more with search term `REST url`. Please note this is very opinionated subject and I don't have pattern on what is right and what is wrong, still:
  - this is Dependency injection test, so resource path `/dependency-injection` instead of `/bean` or something similar would alow you to drop `given...autowired` because it is clear what you are testing
  - you have to thing quite hard to understand what these methods/paths mean, can you simplify it heavily? My examples (find yours!):
    - subpath will be `/spring-person-service` and so full path (with resource path) is `/dependency-injection/spring-person-service` so it is clear that you test dependency injection of SpringPersonService?
    - if you want to test `SpringPersonService#getPersonDao`, you could expose new endpoint and put it to `/dependency-injection/spring-person-service/person-dao`. isn't it bit more readable?

Reply:
Hello,
there is my solution:

- Addressed.
- I tried to make the names shorter and simpler.
- I learned something about REST naming. I applied your suggestions, except the last one, cause it would seem too segmented to me.

Appreciate the helpful suggestions!